FreeRTOS/FreeRTOS-Kernel
# Remove hardware dependence in portmacros.h

Remove hardware dependence in portmacros.h

Description
-----------

The IAR MSP430X port `#include "msp430.h"` which pulls all the hardware register definitions into anything which `#include "FreeRTOS.h"`.  This removes that hardware dependency "leak" by removing the header file and re-defining the `portDISABLE_INTERRUPTS()` and `portENABLE_INTERRUPTS()` macros in terms of `__asm`.
Test Steps

-----------


Checklist:
----------



I tested in IAR at work, don't have a license here at home to test with. Verified by looking at the disassembly of the `portDISABLE_INTERRUPTS()` and `portENABLE_INTERRUPTS()` macros and confirming they are identical to before.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
